subsys/bluetooth/host/classic/a2dp.c

```diff
@@ -76,35 +119,710 @@ struct bt_a2dp *get_new_connection(struct bt_conn *conn)
+	/* Length Of Service Capability */
+	net_buf_add_u8(rsp_buf, ep->codec_cap->len + 2u);
+	/* Media Type */
+	net_buf_add_u8(rsp_buf, seid->sep.media_type << 4u);
+	/* Media Codec Type */
+	net_buf_add_u8(rsp_buf, ep->codec_type);
+	/* Codec Info Element */
+	net_buf_add_mem(rsp_buf, &ep->codec_cap->codec_ie[0], ep->codec_cap->len);
+	return 0;
+}
+
+static int a2dp_set_config_ind(struct bt_avdtp *session, struct bt_avdtp_sep *seid,
+		uint8_t int_seid, struct net_buf *buf, uint8_t *errcode)
+{
+	struct bt_a2dp *a2dp = A2DP_AVDTP(session);
+	struct bt_a2dp_ep *ep;
+	struct bt_a2dp_stream *stream;
+	struct bt_a2dp_stream_ops *ops;
+	uint8_t codec_type;
+	uint8_t *codec_info_element;
+	uint16_t codec_info_element_len;
+	int err;
+
+	*errcode = 0;
+	ep = CONTAINER_OF(seid, struct bt_a2dp_ep, seid);
+	if (ep == NULL) {
+		*errcode = BT_AVDTP_BAD_ACP_SEID;
+		return -1;
+	}
+
+	/* parse the configuration */
+	codec_info_element_len = 4U;
+	err = bt_avdtp_parse_capability_codec(buf,
+				&codec_type, &codec_info_element,
+				&codec_info_element_len);
+	if (err) {
+		*errcode = BT_AVDTP_BAD_ACP_SEID;
+		return -1;
+	}
+
+	if (codec_type == BT_A2DP_SBC) {
+		struct bt_a2dp_codec_sbc_params *sbc_set;
+		struct bt_a2dp_codec_sbc_params *sbc;
+
+		if (codec_info_element_len != 4U) {
+			*errcode = BT_AVDTP_BAD_ACP_SEID;
+			return -1;
+		}
+
+		sbc_set = (struct bt_a2dp_codec_sbc_params *)codec_info_element;
+		sbc = (struct bt_a2dp_codec_sbc_params *)&ep->codec_cap->codec_ie[0];
+		if (((BT_A2DP_SBC_SAMP_FREQ(sbc_set) & BT_A2DP_SBC_SAMP_FREQ(sbc)) == 0) ||
+			((BT_A2DP_SBC_CHAN_MODE(sbc_set) & BT_A2DP_SBC_CHAN_MODE(sbc)) == 0) ||
+			((BT_A2DP_SBC_BLK_LEN(sbc_set) & BT_A2DP_SBC_BLK_LEN(sbc)) == 0) ||
+			((BT_A2DP_SBC_SUB_BAND(sbc_set) & BT_A2DP_SBC_SUB_BAND(sbc)) == 0) ||
+			((BT_A2DP_SBC_ALLOC_MTHD(sbc_set) & BT_A2DP_SBC_ALLOC_MTHD(sbc)) == 0)) {
+			*errcode = BT_AVDTP_BAD_ACP_SEID;
+			return -1;
+		}
+	}
+
+	if ((a2dp_cb != NULL) && (a2dp_cb->config_req != NULL)) {
+		struct bt_a2dp_codec_cfg cfg;
+		struct bt_a2dp_codec_ie codec_config;
+		uint8_t rsp_err_code;
+
+		cfg.codec_config = &codec_config;
+		cfg.codec_config->len = codec_info_element_len;
+		memcpy(&cfg.codec_config->codec_ie[0], codec_info_element,
+			(codec_info_element_len > A2DP_MAX_IE_LENGTH ?
+			A2DP_MAX_IE_LENGTH : codec_info_element_len));
+		err = a2dp_cb->config_req(a2dp, ep, &cfg, &stream, &rsp_err_code);
+		if (err) {
+			*errcode = rsp_err_code;
+		} else if (stream != NULL) {
+			stream->a2dp = a2dp;
+			stream->local_ep = ep;
+			stream->remote_ep_id = int_seid;
+			stream->remote_ep = NULL;
+			stream->codec_config = *cfg.codec_config;
+			ep->stream = stream;
+		}
+	}
+
+	ops = stream->ops;
+	if (*errcode == 0) {
+		if ((ops != NULL) && (stream->ops->configured != NULL)) {
+			stream->ops->configured(stream);
+		}
+	}
+
+	return (*errcode == 0) ? 0 : -1;
+}
+
+#if defined(CONFIG_BT_A2DP_SINK)
+static void bt_a2dp_media_data_callback(
+		struct bt_avdtp_sep *seid,
```

Comment:
I guess the variable name should be renamed too (to `sep`). Same for all other places where you have a `struct avdtp_sep` pointer.

Reply:
Updated